kavm/src/kavm/kavm.py

```diff
@@ -166,19 +167,48 @@ def run_avm_json(
             for app in acc['created-apps']:
                 teal_paths.add(app['params']['approval-program'])
                 teal_paths.add(app['params']['clear-state-program'])
-
-        teal_programs: str = ''
+        try:
+            execute_transactions_stage = [
+                stage for stage in avm_json['stages'] if stage['stage-type'] == 'execute-transactions'
+            ][0]
+        except KeyError:
+            print(f'Test file {input_file} does not contain an "execute-transactions" stage')
+            exit(1)
+        for txn in execute_transactions_stage['data']['transactions']:
+            if 'apap' in txn:
+                teal_paths.add(txn['apap'])
+            if 'apsu' in txn:
+                teal_paths.add(txn['apsu'])
+
+        def run_process_on_bison_parser(path: Path) -> str:
+            print(teal_programs_parser)
+            command = [teal_parser] + [str(path)]
+            res = subprocess.run(command, stdout=subprocess.PIPE, check=True, text=True)
+
+            return res.stdout
+
+        map_union_op = "Lbl'Unds'Map'Unds'{}"
+        map_item_op = "Lbl'UndsPipe'-'-GT-Unds'{}"
+        empty_map_label = "Lbl'Stop'Map{}()"
+        current_teal_pgms_map = empty_map_label
         for teal_path in teal_paths:
-            teal_programs += f'"{teal_path}" |-> {(teal_sources_dir / teal_path).read_text()};'
-        teal_programs += '.TealProgramsStore'
+            teal_path_parsed = 'inj{SortString{},SortKItem{}}(\\dv{SortString{}}("' + str(teal_path) + '"))'
+            teal_parsed = (
+                'inj{SortTealInputPgm{},SortKItem{}}(' + run_process_on_bison_parser(teal_sources_dir / teal_path) + ')'
+            )
+            teal_kore_map_item = map_item_op + '(' + teal_path_parsed + ',' + teal_parsed + ')'
+            current_teal_pgms_map = map_union_op + "(" + current_teal_pgms_map + "," + teal_kore_map_item + ")"
 
         krun_command = ['krun', '--definition', str(self.definition_dir)]
         krun_command += ['--output', output]
-        krun_command += [f'-cTEAL_PROGRAMS={teal_programs}']
-        krun_command += [f'-pTEAL_PROGRAMS={str(teal_programs_parser)}']
+        krun_command += [f'-cTEAL_PROGRAMS={current_teal_pgms_map}']
+        krun_command += ['-pTEAL_PROGRAMS=cat']
```

Comment:
I am confused by the use of `cat` here. If I recall correctly, we have agreed to use the generated parser for `TealProgamsStore` to parse to parse the bundled Teal programs. In fact, after looking at this PR, I agree that we can as well drop the extra `TealProgamsStore` sort and use `Map` with `Id` keys and `TealInputPgm` values (as you are doing now).

We should drop the now unused definition of `TealProgamsStore` from `avm/teal/teal-syntax.md`.

Reply:
`TealProgramsStore` is now removed.